app/controllers/application_controller.rb

```diff
@@ -18,6 +19,10 @@ def current_organization
   end
   helper_method :current_organization
 
+  def current_role
+    session[:current_role] ? Role.find(session[:current_role]) : current_user&.roles&.first
+  end
```

Comment:
Do we need to have them be 1 specific role at a time? Wondering because its another thing to manage and I'am not sure if it is easier without this.

Reply:
Unfortunately yes, because this is how it determines which set of views to show. I may have roles both as a superadmin and an org user, but if I want to see the pages an org user sees, I have to store somewhere the information "I am now pretending to be an org user".